Mage/src/main/java/mage/abilities/effects/common/ChooseACardNameEffect.java

```diff
@@ -86,20 +87,47 @@ public String getChoice(Player player, Game game, Ability source, boolean setVal
     }
 
     private final TypeOfName typeOfName;
+    private final boolean lookAtOpponentHand;
 
     public ChooseACardNameEffect(TypeOfName typeOfName) {
+        this(typeOfName, false);
+    }
+
+    public ChooseACardNameEffect(TypeOfName typeOfName, boolean lookAtOpponentHand) {
         super(Outcome.Detriment);
         this.typeOfName = typeOfName;
-        staticText = "choose " + CardUtil.addArticle(typeOfName.description);
+        this.lookAtOpponentHand = lookAtOpponentHand;
+        if (lookAtOpponentHand) {
+            staticText = "look at an opponent's hand, then choose any " + typeOfName.description;
+        } else {
+            staticText = "choose " + CardUtil.addArticle(typeOfName.description);
+        }
     }
 
     public ChooseACardNameEffect(final ChooseACardNameEffect effect) {
         super(effect);
         this.typeOfName = effect.typeOfName;
+        this.lookAtOpponentHand = effect.lookAtOpponentHand;
     }
 
     @Override
     public boolean apply(Game game, Ability source) {
+        if (lookAtOpponentHand) {
+            Player player = game.getPlayer(source.getControllerId());
+            if (player != null) {
+                TargetOpponent target = new TargetOpponent(true);
+                if (player.choose(Outcome.Benefit, target, source, game)) {
+                    Player opponent = game.getPlayer(target.getFirstTarget());
+                    if (opponent != null) {
+                        MageObject sourceObject = game.getObject(source);
+                        player.lookAtCards(sourceObject != null ? sourceObject.getIdName() : null, opponent.getHand(), game);
+                        player.chooseUse(Outcome.Benefit, "Press Ok to name a card",
+                                "You won't be able to resize the window showing opponents hand once you do",
```

Comment:
This seems like an issue, why can't the window be resized??

Reply:
The pop-up to select a card name is (modal I think is the word).  It doesn't let you click on anything behind it including the "revealed cards" window showing the opponent's hand.

Reply:
I'll open a separate issue for this. The rest of the PR looks good.